test/boost/CMakeLists.txt

```diff
@@ -1,384 +1,200 @@
-add_scylla_test(json_test
-  KIND BOOST
-  LIBRARIES cql3)
-add_scylla_test(keys_test
-  KIND BOOST
-  LIBRARIES idl schema)
-add_scylla_test(large_paging_state_test
-  KIND SEASTAR)
-add_scylla_test(like_matcher_test
-  KIND BOOST
-  LIBRARIES utils)
-add_scylla_test(limiting_data_source_test
-  KIND SEASTAR)
-add_scylla_test(linearizing_input_stream_test
-  KIND BOOST)
-add_scylla_test(lister_test
-  KIND SEASTAR)
-add_scylla_test(loading_cache_test
-  KIND SEASTAR)
-add_scylla_test(locator_topology_test
-  KIND SEASTAR)
-add_scylla_test(log_heap_test
-  KIND BOOST)
-add_scylla_test(logalloc_standard_allocator_segment_pool_backend_test
-  KIND SEASTAR)
-add_scylla_test(logalloc_test
-  KIND SEASTAR)
-add_scylla_test(managed_bytes_test
-  KIND BOOST
-  LIBRARIES Seastar::seastar_testing)
-add_scylla_test(managed_vector_test
-  KIND SEASTAR)
-add_scylla_test(map_difference_test
-  KIND BOOST)
-add_scylla_test(memtable_test
-  KIND SEASTAR)
-add_scylla_test(multishard_combining_reader_as_mutation_source_test
-  KIND SEASTAR)
-add_scylla_test(multishard_mutation_query_test
-  KIND SEASTAR)
-add_scylla_test(murmur_hash_test
-  KIND BOOST)
-add_scylla_test(mutation_fragment_test
-  KIND SEASTAR)
-add_scylla_test(mutation_query_test
-  KIND SEASTAR)
-add_scylla_test(mutation_reader_another_test
-  KIND SEASTAR)
-add_scylla_test(mutation_reader_test
-  KIND SEASTAR)
-add_scylla_test(mutation_test
-  KIND SEASTAR)
-add_scylla_test(mutation_writer_test
-  KIND SEASTAR)
-add_scylla_test(mvcc_test
-  KIND SEASTAR)
-add_scylla_test(network_topology_strategy_test
-  KIND SEASTAR)
-add_scylla_test(nonwrapping_interval_test
-  KIND BOOST)
-add_scylla_test(observable_test
-  KIND BOOST)
-add_scylla_test(partitioner_test
-  KIND SEASTAR)
-add_scylla_test(per_partition_rate_limit_test
-  KIND SEASTAR)
-add_scylla_test(pretty_printers_test
-  KIND BOOST)
-add_scylla_test(querier_cache_test
-  KIND SEASTAR)
-add_scylla_test(query_processor_test
-  KIND SEASTAR)
-add_scylla_test(radix_tree_test
-  KIND SEASTAR)
-add_scylla_test(range_tombstone_list_test
-  KIND BOOST)
-add_scylla_test(rate_limiter_test
-  KIND SEASTAR)
-add_scylla_test(reader_concurrency_semaphore_test
-  KIND SEASTAR)
-add_scylla_test(recent_entries_map_test
-  KIND SEASTAR)
-add_scylla_test(repair_test
-  KIND SEASTAR)
-add_scylla_test(restrictions_test
-  KIND SEASTAR)
-add_scylla_test(result_utils_test
-  KIND SEASTAR)
-add_scylla_test(reusable_buffer_test
-  KIND SEASTAR)
-add_scylla_test(role_manager_test
-  KIND SEASTAR)
-add_scylla_test(row_cache_test
-  KIND SEASTAR)
-add_scylla_test(rust_test
-  KIND BOOST
-  LIBRARIES inc)
-add_scylla_test(s3_test
-  KIND SEASTAR)
-add_scylla_test(aws_errors_test
-  KIND BOOST)
-add_scylla_test(aws_error_injection_test
-  KIND SEASTAR)
-add_scylla_test(schema_change_test
-  KIND SEASTAR)
-add_scylla_test(schema_changes_test
-  KIND SEASTAR)
-add_scylla_test(schema_loader_test
+add_scylla_test(boost_combined_test
+  SOURCES boost_combined_test.cc
```

Comment:
both `SOURCES` and `LIBRARIES` are so called multi-value arguments. see https://github.com/scylladb/scylladb/blob/3fda9651ccbcb7b23f843515297683bca091253b/test/CMakeLists.txt#L11 . and we use [cmake_parse_arguments](https://cmake.org/cmake/help/latest/command/cmake_parse_arguments.html) to parse the arguments. in general, one can just use something like
```cmake
add_scylla_test(boost_combined_test
  SOURCES
    boost_combined_test.cc
    UUID_test.cc
    # more sources here
  LIBRARIES
    auth
    alternator
    # more libraries here
    Seastar::seastar_testing
  KIND BOOST)
```
to define a test.

in which, i'd suggest 

- add indent to the arguments passed to the keywords like `SOURCES`
- order the source files alphabetically
- group the external libraries and internal/convenience libraries separately